File: Python/count-and-say.py
class Solution:
    def countAndSay(self, n):
        """
        :type n: int
        :rtype: str
        """
        if n <= 0:
            return ""

        res = "1"

        for _ in range(1, n):
            temp = ""
            count = 1
            chars = list(res)
            current = chars[0]

            for i in range(1, len(chars)):
                if chars[i] == current:
                    count += 1
                else:
                    temp += str(count)
                    temp += current
                    count = 1
                    current = chars[i]

            temp += str(count)
            temp += current
            res = temp

        return res

File: Swift/count-and-say.swift
class Solution {
    func countAndSay(_ n: Int) -> String {
        if n <= 0 {
            return ""
        }
        
        var res = "1"
        
        for _ in 1..<n {
            var temp = ""
            var count = 1
            var chars = [Character](res)
            var current = chars[0]
            
            for i in 1..<chars.count {
                if chars[i] == current {
                    count += 1
                } else {
                    temp.append("\(count)")
                    temp.append(current)
                    count = 1
                    current = chars[i]
                }
            }
            temp.append("\(count)")
            temp.append(current)
            res = temp
        }
        
        return res
    }
}
